packages/ai/core/generate-object/output-strategy.ts

```diff
@@ -290,7 +290,7 @@ const arrayOutputStrategy = <ELEMENT>(
 
 const enumOutputStrategy = <ENUM extends string>(
   enumValues: Array<ENUM>,
-): OutputStrategy<ENUM, ENUM, never> => {
+): OutputStrategy<ENUM | (string & {}), ENUM, never> => {
```

Comment:
why is this needed?

Reply:
enums partial streaming can either return the full enum `foobar` or a prefix like `foo` (-`bar`) so I added a partial type to make autocomplete still work will making sure there's a `string` | type

Reply:
I don't think this is what the users would expect - then they'd need to handle the string case. I would prefer a solution that is closer to what we do for arrays - i.e., only return ENUM values even as partials.

Reply:
wonder if we can just return `string` then

Reply:
done :pray: